### PR TITLE
Close #2661 - add no-compile flag

### LINKED ISSUE #2661: add no-compile flag
## Issue

It would be nice to have a `--no-compile` flag that skips compilation on commands that otherwise include it by default. This would also solve the issue here: https://github.com/trufflesuite/truffle/issues/469.

This is a command I needed and had to switch to buidler to use because I have some contracts which for legacy purposes must remain tied to solc4 and some contracts which should use solc5. In order to compile them I need to set a different config file for each compilation. Then when it comes to migrating or testing them, I want to prevent them from being compiled with the wrong compiler, so i include the `--no-compile` option.

### PR DESCRIPTION
**Pull request by robertmagier**
I throw an error when flags --compile-all or --debug are used together with -compile-none
Not sure if this is the correct way to do it. 
```

    if (config.compileAll && config.compileNone) {
      options.logger.log(
        "--compile-none can not be set together with --compile-all or --debug"
      );
      throw "Command line error.";
    }
```

### CODE REVIEW

File: packages/core/lib/test.js [hunk 1]
```diff
@@ -98,26 +98,31 @@ const Test = {
...
-      return testResolver.require(testFilePath);
-    });
-
     const runner = new TestRunner(config);
+    var compilations = null;
+
+    // Do not compile contracts, test contracts, do not add test contracts to testing set
+    // when --compile-none flag is set
+    if (!config.compileNone) {
+      var { compilations } = await this.compileContractsWithTestFilesIfNeeded(
```
Comment (gnidan) [inline packages/core/lib/test.js]: Instead of declaring `compilations` twice, could you move this `compileNone` check to inside the `compileContractsWithTestFilesIfNeeded` method?

Comment (robertmagier) [reply packages/core/lib/test.js]: Ok. I can return {} if config.compileNone is set to true. Does it make sense ? 
I will still have to make check for config.compileNone flag in Test.run function to skip solidity tests if this flag is set.

**Comment by robertmagier** [comment]
@gnidan Should I add test case for --compile-none flag scenario ?

**Comment by robertmagier** [comment]
I added test file to test how to use --compile-none flag. In last scenario this test is failing exactly as in issue #469 Can someone verify if this scenario is correct ?

**Comment by robertmagier** [comment]
@gnidan  @CruzMolina Pleeeease ;)

**Comment by gnidan** [comment]
Ended up taking a different approach, but this should be done now as of #3208.

Sorry for the radio silence here, but I think this PR can now be closed, since that functionality now exists!

**Comment by robertmagier** [comment]
All my work for nothing.

**Comment by gnidan** [comment]
I'm really sorry about that @robertmagier! I felt awful but still decided that the alternate approach was simple enough.

Looking at this again now, however, I think your tests will provide lots of value! I'll reopen this with the plan to get these tests merged, at least!

**Comment by eggplantzzz** [comment]
Some of this work was taken and used in https://github.com/trufflesuite/truffle/pull/3335. I gave coauthoring credit to @robertmagier.

**Comment by robertmagier** [comment]
I am sorry for whining ;) Thank you for giving coauthoring credit to me as well. Let me know if you need any help here... or there.

**Comment by eggplantzzz** [comment]
It's no problem :) I hate to have good work go to waste, it has been a busy time the last year or so but we have implemented some process to help us to stay on top of contributions like this one. Is your Windows CI PR still relevant?

**Comment by eggplantzzz** [comment]
So this feature was merged in [this PR](https://github.com/trufflesuite/truffle/pull/3335) and should be released next week!

**Comment by eggplantzzz** [comment]
Closing as this feature was released in version 5.1.47. Thanks!
